### PR TITLE
feat: Add optional custom button for adornment

### PR DESCRIPTION
**Pull request by jbruce1-ocrolus**
Add support for custom button in adornment as well controlling position of adornment on left or right hand of input with affecting header text.

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -26,6 +27,7 @@ export type {
   MuiColorInputProps,
   MuiColorInputValue,
   MuiColorInputColors,
+  ColorButtonProps,
```
Comment (viclafouch) [inline src/index.tsx:30]: MuiColorButtonProps

File: src/index.tsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ import {
 import { assocRefToPropRef } from '@shared/helpers/ref'
 
 import type {
+  ColorButtonProps,
```
Comment (viclafouch) [inline src/index.tsx:18]: MuiColorButtonProps

File: src/index.types.ts [hunk 3]
```diff
@@ -29,6 +33,8 @@ export type MuiColorInputColors = {
 
 export interface MuiColorInputProps extends TextFieldProps {
   value: MuiColorInputValue
+  adornmentPosition: 'start' | 'end'
```
Comment (viclafouch) [inline src/index.types.ts:36]: optional ?

**Comment by jbruce1-ocrolus** [comment]
@viclafouch resolved those comments

File: src/index.types.ts [hunk 4]
```diff
@@ -1,5 +1,9 @@
 import type { PopoverProps as MuiPopoverProps } from '@mui/material/Popover'
 import type { TextFieldProps as MuiTextFieldProps } from '@mui/material/TextField'
+import {
```
Comment (viclafouch) [inline src/index.types.ts:3]: `import type` (sorry didn't see)

File: src/components/ColorPopover/ColorPopover.tsx [hunk 5]
```diff
@@ -4,18 +4,23 @@ import Popover, { PopoverProps } from '@mui/material/Popover'
 import { Styled } from './ColorPopover.styled'
 
 type ColorPopoverProps = PopoverProps & {
+  position: 'start' | 'end'
```
Comment (viclafouch) [inline src/components/ColorPopover/ColorPopover.tsx:7]: I think it can be optional too

**Comment by jbruce1-ocrolus** [comment]
@viclafouch alright third times the charm

**Comment by viclafouch** [comment]
I merge this asap ! ;)

**Pull request merged**